### PR TITLE
CORE-1450 Fix Step.Config.Parameters sort order

### PR DESCRIPTION
**Pull request by psarando**
This PR will fix the `Step.Config.Parameters` sort order by using a stable sort for sorting params, which also fixes the `Arguments` order used for the command-line.

This is particularly important for multi-input params, which can all have the same param order, but only the 1st will have an argument option ("name") which can't be shuffled into the middle of the list of inputs.

For example, using the vanilla sort function could produce a command line like
`{"/usr/local2/bin/QATestTool.sh", "param1", "Acer-tree.txt", "param3", "true", "param0", "wc_out.txt", "param4", "four", "input.3", "input.4", "input.5", "input.6", "--multi-param2", "input.1", "input.2"}`
instead of
`{"/usr/local2/bin/QATestTool.sh", "param1", "Acer-tree.txt", "param3", "true", "param0", "wc_out.txt", "param4", "four", "--multi-param2", "input.1", "input.2", "input.3", "input.4", "input.5", "input.6"}`

This PR also switches the repo to Go modules.

### CODE REVIEW

File: step.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ type StepConfig struct {
 // Parameters returns the StepParams associated with a Step in the correct order.
 // Use this to get the list of Params rather than accessing the field directory.
 func (c *StepConfig) Parameters() []StepParam {
-	sort.Sort(ByOrder(c.Params))
+	sort.Stable(ByOrder(c.Params))
```
Comment (psarando) [inline step.go:146]: This is the actual fix.

**Review by slr71 — COMMENTED** [review]
✨ Looks good to me! :shipit:

**Review by slr71 — APPROVED** [review]
I forgot to mark my last review as an approval. 😖

**Comment by psarando** [comment]
Thanks for the review 👍 

Thought now I'm not sure if the Dockerfile needs to be updated 🤔

**Review by johnworth — APPROVED** [review]
LGTM 🚢🚚

**Comment by psarando** [comment]
Thanks for the reviews! 👍

**Pull request merged**